src/button-group/internal.tsx

```diff
@@ -148,43 +148,44 @@ const InternalButtonGroup = forwardRef(
           getNextFocusTarget={getNextFocusTarget}
           onUnregisterActive={onUnregisterActive}
         >
-          {items.map((itemOrGroup, index) => {
-            const itemContent = (item: ButtonGroupProps.Item, position: string) => (
-              <ItemElement
-                key={item.id}
-                item={item}
-                dropdownExpandToViewport={dropdownExpandToViewport}
-                tooltip={tooltip}
-                setTooltip={setTooltip}
-                onItemClick={onItemClick}
-                onFilesChange={onFilesChange}
-                ref={element => (itemsRef.current[item.id] = element)}
-                position={position}
-              />
-            );
-
-            const isGroupBefore = items[index - 1]?.type === 'group';
-            const currentItem = items[index];
-            const isGroupNow = currentItem?.type === 'group';
-            const shouldAddDivider = isGroupBefore || (!isGroupBefore && isGroupNow && index !== 0);
-
-            if (isGroupNow && currentItem.items.length === 0) {
-              warnOnce('ButtonGroup', 'Empty group detected. Empty groups are not allowed.');
-            }
-
-            return (
-              <React.Fragment key={itemOrGroup.type === 'group' ? index : itemOrGroup.id}>
-                {shouldAddDivider && <div className={styles.divider} />}
-                {itemOrGroup.type === 'group' ? (
-                  <div key={index} role="group" aria-label={itemOrGroup.text} className={styles.group}>
-                    {itemOrGroup.items.map((item, subIndex) => itemContent(item, `${index + 1},${subIndex + 1}`))}
-                  </div>
-                ) : (
-                  itemContent(itemOrGroup, `${index + 1}`)
-                )}
-              </React.Fragment>
-            );
-          })}
+          {variant === 'children'
```

Comment:
I think that we need to allow both items and custom items simultaneously. If decided otherwise, I recommend adding a dev warning if both items and children props are used (or, rather if using "icon" variant with no items or "children" variant with no childen).